vector.hpp

```diff
@@ -0,0 +1,322 @@
+
+/*
+ * Author: Branislav Pilnan, Uday Patel
+ * Organisation: HYPED
+ * Date: 17 February 2018
+ * Description: K-dimensional vector class that supports 
+ *              addition, substraction, scalar multiplication,
+ *              scalar division.
+ *
+ *    Copyright 2018 HYPED
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <array>
+
+#ifndef VECTOR_HPP_
+#define VECTOR_HPP_
+
+template <typename T, int vector_size>
+class Vector
+{
+public:
+  
+  /**
+   * @brief    Constructors for the class for a zero vector.
+   */
+  Vector();
+
+  /**
+   * @brief    Constructors for the class for a particular vector.
+   */
+  Vector(const T vector[]);
+
+  /**
+   * @brief    Conversion from a vector type to another.
+   */
+  Vector(const Vector<int,vector_size> &rhs);
+  Vector(const Vector<float,vector_size> &rhs);
+  Vector(const Vector<double,vector_size> &rhs);
```

Comment:
I don't see a point in supporting conversion for some types but not other types. We should rather not do it at all.